controllers/front/PostComment.php

```diff
@@ -0,0 +1,150 @@
+            ->setContent($comment_content)
+            ->setCustomerName($customer_name)
+            ->setCustomerId($this->context->cookie->id_customer)
+            ->setGuestId($this->context->cookie->id_guest)
+            ->setDateAdd(new \DateTime('now', new \DateTimeZone('UTC')))
+        ;
+        $entityManager->persist($productComment);
+        $this->addCommentGrades($productComment, $criterions);
+
+        //Validate comment
+        if (!empty($errors = $this->validateComment($productComment))) {
+            $this->ajaxRender(json_encode([
+                'success' => false,
+                'errors' => $errors,
+            ]));
+
+            return false;
+        }
+
+        $entityManager->flush();
+
+        $this->ajaxRender(json_encode([
+            'success' => true,
+            'product_comment' => $productComment->toArray(),
+        ]));
+    }
+
+    /**
+     * @param ProductComment $productComment
+     * @param array $criterions
+     *
+     * @throws Exception
+     */
+    private function addCommentGrades(ProductComment $productComment, array $criterions)
+    {
+        /** @var EntityManagerInterface $entityManager */
+        $entityManager = $this->container->get('doctrine.orm.entity_manager');
+        $criterionRepository = $entityManager->getRepository(ProductCommentCriterion::class);
+        $averageGrade = 0;
+        foreach ($criterions as $criterionId => $grade) {
+            $criterion = $criterionRepository->findOneById($criterionId);
+            $criterionGrade = new ProductCommentGrade(
+                $productComment,
+                $criterion,
+                $grade
+            );
+            $entityManager->persist($criterionGrade);
+            $averageGrade += $grade;
+        }
+        $averageGrade /= count($criterions);
+        $productComment->setGrade($averageGrade);
+    }
+
+    /**
+     * Manual validation for now, this would be nice to use Symfony validator with the annotation
+     *
+     * @param ProductComment $productComment
+     *
+     * @return array
+     */
+    private function validateComment(ProductComment $productComment)
+    {
+        $errors = [];
+        if (empty($productComment->getTitle())) {
+            $errors[] = $this->trans('Title cannot be empty', [], 'Modules.Productcomments.Shop');
+        } elseif (strlen($productComment->getTitle()) > 64) {
+            $errors[] = $this->trans('Title cannot be more than %s characters', [64],'Modules.Productcomments.Shop');
+        }
+
+        if (!$productComment->getCustomerId()) {
+            if (empty($productComment->getCustomerName())) {
+                $errors[] = $this->trans('Customer name cannot be empty', [], 'Modules.Productcomments.Shop');
+            } elseif (strlen($productComment->getCustomerName()) > 64) {
+                $errors[] = $this->trans('Customer name cannot be more than %s characters', [64], 'Modules.Productcomments.Shop');
```

Comment:
Samme comment about having a const for the limit of characters for customers.

Reply:
Done